### DISCUSSION TITLE
Welcome to CloverBootloader Discussions!

### DISCUSSION THREAD

**Discussion by SergeySlice** [CloverHackyColor/CloverBootloader#324] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jief666** [comment]
Hi,
I'm Jief. Developer since a very long time. I'm using Clover and slowly refactor the code to make it easier to read.
I'm more high level programming (OOP design) and env setup guy (IDE/Live debugging/Virtual test machine).
I think it's good to separate discussions instead of having all in one thread. I can't follow all what is said on Insanelymac...
Merry Christmas all.

**Comment by chris1111** [comment]
Yes I think the same thing more everything is done quickly here
Merry Christmas all and Happy New Years

**Comment by serdeliuk** [comment]
Happy New Year!!!

**Comment by w4601251** [comment]
Can I have a Chinese debugging manual? I thought that I had just started to contact with the official debugging manual. I couldn't understand the language problem. thank you

**Comment by w4601251** [comment]
Or you can make a web page to explain the contents of the debugging manual.

It is also convenient for you to translate directly with Google

**Comment by glnzglnz** [comment]
Sorry I'm such a noob, but I don't know my way around github, and I don't know what to download (or then do) in order to install Clover onto a bootable USB stick.
How do I download which files, and then what should I do to create my Clover USB stick?  My goal is to use Clover to get my Dell Optiplex 7010 to boot from a NVMe SSD on an adapter card in a PCIe slot.  The SSD will (eventually) hold a clone of my current Win 10 Pro 64-bit HDD,
Happy to take any links that show how to create the Clover stick from these pages.
Thanks, and Happy Holidays.

**Reply by Morbius01** [reply]
Well, Clover should enable you to do exactly what you are looking for.
See my post [here](https://github.com/CloverHackyColor/CloverBootloader/discussions/491)

Have you succeeded in creating your Clover key, and then in installing your OS on your NVMe SSD ?
If not, I might be able to help you if you are still interested...

**Comment by glnzglnz** [comment]
Morbius - thanks for reply.  I have not yet tried but am still interested.  In my case, I have a Dell Optiplex 7010 Mini-Tower, and my intention is to clone my existing Win 10 Pro 64-bit into an NVMe M.2 in an adapter card in a PCIe slot and be able to boot from that.. 

Problem is that that github is not for a noob such as myself - there is no step-by-step as to which files to download or how to install them onto the USB stick, or then how to use the resulting Clover app.. 

For example, on a different thread in these Discussions, someone said that he had to set his native BIOS back to Legacy to get Clover to work - very unintuitive.  

Will be happy to get some basic steps.  Thanks..

**Reply by Morbius01** [reply]
I have not seen anywhere a real step-by-step guide for installing Clover with detailed instructions.
There is a large thread on the Win-Raid forum [here](https://www.win-raid.com/t2375f50-Guide-NVMe-boot-w-o-BIOS-modding-Clover-EFI-bootloader-method.html) which you probably know since I have seen you (or someone with the same name !) posting somewhere in the last pages… But this thread is too large (~55 pages now) and too much time consuming to read in its entirety...

It took me almost 1 month before I succeeded in creating a Clover key which works, and then installing several OSs on my non-bootable NVMe SSD, and boot on them through Clover.
I believe that someone repeating the steps I followed can do the same in less that 3 hours.
And I would be happy to share my knowledge.

I am not sure what you mean by “cloning” your current Win 10.
All I can explain you with step-by-step instructions is how to:
1. Create a Clover key.
2. Install a fresh version of Win 10 on your NVMe M.2 SSD using your Win 10 installation disk (or USB drive). You will then be able to boot on this Win 10 installed on your SSD through Clover.

You will then have to restore all your data, which supposes that you will have previously backed them up from your current Win 10.

And, by the way, don’t worry about “...set your BIOS back to Legacy…”.
All you need to do is to make sure that, when you have a USB drive inserted in your system, your system will boot from this USB drive (primary boot device), and not from another device such as a HDD. I am pretty sure you will know how to do this.

Let me know if you are interested, and I will provide you help on how to achieve these 2 steps.
